package.json

```diff
@@ -50,7 +50,9 @@
     "examples": "node ./examples/runner.js",
     "integration_short": "./node_modules/.bin/mocha test/integration/short -R spec -t 5000 --recursive --exit",
     "integration_long": "./node_modules/.bin/mocha test/integration/long -R spec -t 5000 --recursive --exit",
-    "pre-push": "npx prettier \"{lib,examples,test}/**/*.js\" --write && cargo fmt && cargo clippy --all-targets --all-features -- -D warnings",
+    "prettier": "npx prettier \"{lib,examples,test}/**/*.js\" --write",
+    "full-clippy": "cargo clippy --all-targets --all-features -- -D warnings",
+    "pre-push": "npm run prettier && cargo fmt && npm run full-clippy && napi build --platform",
```

Comment:
I understand that `pre-push` is something that a contributor should invoke themselves before commiting. However, is running it enforced? Does CI guard against not running it?

Reply:
``Code quality`` CI step checks if all 4 steps are satisfied:
- cargo clippy 
- cargo fmt (just checking)
- node prettier (just checking)
- ensuring ``index.js`` and ``index.d.ts`` are based on current code version

Reply:
Are you sure that running `napi build --platform` actually verifies that the generated code is up-to-date? I suspect that the CI does not commit any changes, so even if it builds up-to-date files when the pushed ones were outdated, it does not solve the problem of outdated files on a checked branch - those are there still and the check passes.

Reply:
I'm not sure I understand your concern here.

Reply:
Nvm, I missed your updates of `code-quality.yml` workflow that indeed ensure that `index.js` and `index.d.ts` are up-to-date.